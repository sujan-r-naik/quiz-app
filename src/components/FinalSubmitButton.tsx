import { Box, Button } from "@mui/material";
import { useNavigate } from "react-router-dom";
import { matchq } from "../constants/Questions";

const FinalSubmitButton = (props: { data: any[]; matchData: any }) => {
  const navigate = useNavigate();

  const handleSub = () => {
    if (props.data.length !== 0) {
      let ans = props.data.filter((x) => Object.keys(x).length !== 0);

      var trues = 0;
      for (let i = 0; i < ans.length; i++) {
        if (ans[i].a[0] === ans[i].ca[0]) {
          trues++;
        }
      }

      if (
        Object.entries(props.matchData).sort().toString() ===
        Object.entries(matchq).sort().toString()
      )
        trues++;

      let len = `${trues}`;
      ans[4] = {
        qi: 4,
        q: "Match the Following",
        a: [
          {
            answerText: matchq,
            isCorrect: true,
          },
        ],
        ca: [
          {
            answerText: props.matchData,
            isCorrect:
              Object.entries(props.matchData).sort().toString() ===
              Object.entries(matchq).sort().toString(),
          },
        ],
      };
      console.log(ans);
      navigate("/score", {
        state: {
          trues: trues,
          ans: ans,
        },
      });
    }
  };

  return (
    <Box
      role="fsubHolder"
      sx={{
        height: "10vh",
        width: "100vw",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Button role="fsub" sx={{ alignSelf: "center" }} onClick={handleSub}>
        Final Submit
      </Button>
    </Box>
  );
};

export default FinalSubmitButton;
