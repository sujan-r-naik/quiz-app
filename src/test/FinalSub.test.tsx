import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { screen, render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import FinalSubmitButton from "../components/FinalSubmitButton";

test("Snapshot of Home component", () => {
  const comp = renderer.create(
    <BrowserRouter>
      <FinalSubmitButton data={[]} matchData={[{}]} />
    </BrowserRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Home,", () => {
  test("does the sub components of finalSubmit work", () => {
    render(
      <BrowserRouter>
        <FinalSubmitButton data={[]} matchData={[]} />
      </BrowserRouter>
    );

    let a = screen.getByRole("fsub");
    let b = screen.getByRole("fsubHolder");
    expect(a).toBeInTheDocument();
    expect(b).toBeInTheDocument();
  });

  test("does fn work", () => {
    render(
      <BrowserRouter>
        <FinalSubmitButton
          data={[
            {
              qi: 4,
              q: "",
              a: [
                {
                  answerText: "",
                  isCorrect: true,
                },
              ],
              ca: [
                {
                  answerText: "",
                  isCorrect: true,
                },
              ],
            },
          ]}
          matchData={[{}]}
        />
      </BrowserRouter>
    );

    let btn = screen.getByRole("fsub");
    expect(btn).toBeInTheDocument();

    let a = fireEvent.click(btn);
    expect(a).toBeTruthy();
  });
});
