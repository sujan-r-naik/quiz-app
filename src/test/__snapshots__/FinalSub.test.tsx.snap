// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Home component 1`] = `
<div
  className="MuiBox-root css-1tqsx7z"
  role="fsubHolder"
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1ir790h-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="fsub"
    tabIndex={0}
    type="button"
  >
    Final Submit
  </button>
</div>
`;
